src/main/java/model/Car.java

```diff
@@ -0,0 +1,115 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Car {
+    private static final boolean WINNER = true;
+    private static final boolean LOSER = false;
+    private static final int DRIVE_POSITION = 1;
+    private static final int CONDITION_MIN_VALUE = 0;
+    private static final int CONDITION_MAX_VALUE = 10;
+    private static final int CONDITION_GOOD_MIN_VALUE = 4;
+    private static final boolean GOOD = true;
+    private static final boolean BAD = false;
+    private static final int CORRECT_NAME_LENGTH_LIMIT = 5;
+    private final String name;
+    private int position = 0;
+
+    public Car(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getPosition() {
+        return position;
+    }
+
+    private boolean isNotBlank(String name) {
+        return !name.trim().isEmpty();
+    }
+
+    private boolean isLengthIsBelowCorrectNameLengthLimit(String name) {
+        return name.trim().length() <= CORRECT_NAME_LENGTH_LIMIT;
+    }
+
+    public boolean isCorrectName() {
+        return isNotBlank(name) && isLengthIsBelowCorrectNameLengthLimit(name);
+    }
+
+    public List<Car> setUpCars(List<Car> cars) {
+        Car car = new Car(name.trim());
+        cars.add(car);
+        return cars;
+    }
+
+    public int createConditionValue() {
+        int conditionValue = (int) (CONDITION_MIN_VALUE + Math.random() * CONDITION_MAX_VALUE);
+        return conditionValue;
+    }
+
+    private boolean isGood(int conditionValue) {
+        if (conditionValue >= CONDITION_GOOD_MIN_VALUE) {
+            return GOOD;
+        }
+        return BAD;
+    }
+
+    public static List<Car> setCars(String input) {
+        List<Car> cars = new ArrayList<>();
+        String[] names = input.split(",");
+        boolean isCorrectNames = Arrays.stream(names)
+            .map(Car::new)
+            .allMatch(car -> car.isCorrectName());
+        if (isCorrectNames) {
+            Arrays.stream(names)
+                .map(Car::new)
+                .forEach(car -> car.setUpCars(cars));
+        }
+        return cars;
+    }
+
+    public void drive(int conditionValue) {
+        if (isGood(conditionValue)) {
+            this.position = this.position + DRIVE_POSITION;
+        }
+    }
+
+
+    public static void proceed(List<Car> cars) {
+        int size = cars.size();
+        for (int i = 0; i < size; i++) {
+            Car car = cars.get(i);
+            int conditionValue = car.createConditionValue();
+            car.drive(conditionValue);
```

Comment:
```suggestion
            car.drive(car.createConditionValue());
```
이렇게 변경해도 문제가 될 것 같지 않습니다. 이렇게 코드를 변경하고 보니까 ``createConditionValue()`` 부분이 car에서 호출되고 있는 것이 조금 이상하다는 생각이 들지 않나요?

Reply:
자동차 객체 자체가 갖는 랜덤한 값인 conditionValue는 자동차가 자율적으로 만들어내고 스스로 판단하여 전진할지 정지할지 정해야 한다고 생각이 됩니다. createConditionValue 대신 getConditionValue로 메서드명을 달리 하였다면 다르게 느껴지셨을지 궁금합니다. 그리고 방금 말씀드린 제 견해에 대한 생각도 듣고 싶습니다.